### PR TITLE
Rfc008 - Certificate structure

### PR DESCRIPTION
**Pull request by woutslakhorst**
Missing img:

certificates_v1_alt

### CODE REVIEW

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: rfc/rfc008-certificate-structure.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1.  Introduction
+
+In a secure network certificates play an important role. They are required to set up a TLS connection, and they provide a mechanism for establishing trust in published data.
+
+When it comes to certificates, we’d like to be at the edge of usability and security. All administrative actions regarding keys should be simple and logical to the administrator and yet it should follow the latest security standards. A very elaborate certificate tree structure covers all requirements but is too complicated to understand. Complication can lead to human error, which in the case of private key management leads to security holes. Apart from supporting the production use-case, we also want to make things easier for development and demonstrations. This means that the chosen solutions supports a wide variety of tools and/or scripting.
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply rfc/rfc008-certificate-structure.md:31]: changes solutions -> solution

File: rfc/rfc008-certificate-structure.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+|  |  |
+| :--- | :--- |
+| Nuts foundation | W.M. Slakhorst |
+| Request for Comments: 003 | Nedap |
+|  | R.G. Krul |
+|  | Nedap |
+|  | Oktober 2020 |
+
+## Certificate Structure
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply rfc/rfc008-certificate-structure.md:11]: We'll add the details for the data TLS connection to this RFC when it's ready. Main reason is that supporting the current rfc8 is doable for production but supporting the data TLS connection with client certificates is more problematic. Meaning we have do not comply with the rfc for a long time.

File: rfc/rfc008-certificate-structure.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+## 1.  Introduction
+
+In a secure network certificates play an important role. They are required to set up a TLS connection, and they provide a mechanism for establishing trust in published data.
+
+When it comes to certificates, we’d like to be at the edge of usability and security. All administrative actions regarding keys should be simple and logical to the administrator and yet it should follow the latest security standards. A very elaborate certificate tree structure covers all requirements but is too complicated to understand. Complication can lead to human error, which in the case of private key management leads to security holes. Apart from supporting the production use-case, we also want to make things easier for development and demonstrations. This means that the chosen solutions supports a wide variety of tools and/or scripting.
+
+## 2. Terminology
+
+* **CA**: Certificate authority as defined by [RFC2459](https://tools.ietf.org/html/rfc2459).
+* **document log**: data structure for the shared network state.
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply rfc/rfc008-certificate-structure.md:36]: no longer needed

File: rfc/rfc008-certificate-structure.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+* **shared network state**: the accumulation of events or document log entries that are in the right order and where every node agrees upon.
+* **truststore**: The file that contains the trusted CA's.
+
+Other terminology is taken from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Overview
+
+insert picture here
+
+The figure above shows the vendor CA tree. The lines indicate that a lower certificate is signed by a higher one in the graph. As can be seen, the vendor CA is the root of the tree. Each vendor MUST generate a self-signed certificate and have it published by another node. An organization proof SHOULD be published in the same event linking the certificate to the real world organization. See [XXX] for an explanation on the organization proof. A vendor CA MAY still be trusted by another node manually, this would involve loading the generated document log from node A into node B.
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply rfc/rfc008-certificate-structure.md:49]: Left it to RFC006 to explain the details

File: rfc/rfc008-certificate-structure.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+| Extended Key Usage 	| 
+serverAuthentication, clientAuthentication 	|
+| SubjectAlternativeName 	| otherName=1.3.6.1.4.1.54851.4:<vendor ID> 	|
+
+## 5. Private key algorithms
+
+We’re currently settling on the secp384r1 elliptic curve. 384 bits is one step above 256 which should already give 10 years of protection. When key rotation is set up from the beginning, we can change the desired key algorithm easily. A specific command SHOULD be available on a node to help the administrator achieve this. Some diagnostics on when a key has to be rotated COULD be provided as well.
+
+## 6. Private key management
+The following paragraphs are more of a guideline than a hard requirement since it involves the internal workings of a node.
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply rfc/rfc008-certificate-structure.md:111]: Not sure, I think we'll have to reevaluate this RFC when implementing it.

File: rfc/rfc008-certificate-structure.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+* **document log**: data structure for the shared network state.
+* **event type**: the type of entry in the document log.
+* **node**: a running instance of a Nuts implementation (a.k.a. Nuts node).
+* **node operator**: an organization running a node. Typically a vendor that creates healthcare software for care organizations.
+* **shared network state**: the accumulation of events or document log entries that are in the right order and where every node agrees upon.
+* **truststore**: The file that contains the trusted CA's.
+
+Other terminology is taken from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Overview
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply rfc/rfc008-certificate-structure.md:45]: Added some more intro text

**Review by reinkrul — APPROVED** [review]

**Pull request merged**